### ISSUE TITLE
Could not generate `toJson` code for `subjects` because of type Entity, (@Flutter, @JsonSerializable, @Dart)

### ISSUE THREAD

**Issue by khursheed33** [google/json_serializable.dart#1159]
Flutter: Not able to use nested models with Entities. (Entities, Models, Equatable, Flutter, Dart, json_serializable,Json Serialization)

I have two file for a single data i.e Entity and Model. To separate the entire logic from each other and to follow best practices. Whenever I am trying to use nested Entities and parsing through @JsonSerializable it's showing me error.

Note: I have tried @JsonConverter too. It's not working.

I'm sure I'm missing something over there but can't get it. 
Please help.

SEE:
https://stackoverflow.com/q/72668678/10872954

**Comment by coccyx-cyst** [comment]
@khursheed33 You could try to override your Model field like so:

```dart
@JsonSerializable(explicitToJson: true)
class TestModel extends TestEntity {

@override
final List<AddressModel> address;

const TestModel({
required String id,
required List<AddressModel> address,
}) : super(
      id: id,
      address: address,
    );
factory TestModel.fromJson(Map<String, dynamic> json) =>
  _$TestModelFromJson(json);

Map<String, dynamic> toJson() => _$TestModelToJson(this);
}
```

**Comment by khursheed33** [comment]
Exactly that's what I did, thanks for your response.

**Comment by ViktorKirjanov** [comment]
> @khursheed33 You could try to override your Model field like so:
> 
> ```dart
> @JsonSerializable(explicitToJson: true)
> class TestModel extends TestEntity {
> 
> @override
> final List<AddressModel> address;
> 
> const TestModel({
> required String id,
> required List<AddressModel> address,
> }) : super(
>       id: id,
>       address: address,
>     );
> factory TestModel.fromJson(Map<String, dynamic> json) =>
>   _$TestModelFromJson(json);
> 
> Map<String, dynamic> toJson() => _$TestModelToJson(this);
> }
> ```

its wrong, you will get an error:
All final variables must be initialized, but 'address' isn't.

**Comment by rafaeljustinox** [comment]
I don't think this is the right answer because it makes the class twice as big in memory. if you analyze it in the debugger, you'll see that the class has duplicate properties.

**Comment by feduke-nukem** [comment]
> I don't think this is the right answer because it makes the class twice as big in memory. if you analyze it in the debugger, you'll see that the class has duplicate properties.

You are welcome to introduce right answer
